lib/fs.js

```diff
@@ -1754,6 +1758,15 @@ function symlink(target, path, type_, callback_) {
   const type = (typeof type_ === 'string' ? type_ : null);
   const callback = makeCallback(arguments[arguments.length - 1]);
 
+  if (permission.isEnabled()) {
+    // The permission model's security guarantees fall apart in the presence of
+    // relative symbolic links. Thus, we have to prevent their creation.
+    if (typeof target !== 'string' || !isAbsolute(toPathIfFileURL(target))) {
```

Comment:
I think we could check if it's a Buffer and default it to utf8 to check if `isAbsolute`.

Reply:
On POSIX, that generally works because POSIX only checks the first byte (not character) of the path to see if it is absolute. On Windows, I am not so sure.

Reply:
I have played a bit on Windows today (I was fixing another issue) and I couldn't reproduce the case you've mentioned.

Reply:
@RafaelGSS Is this a blocking suggestion? I think it'd be appropriate to add support for `Uint8Array` paths in a follow-up PR if desired. I myself am not confident that I'd do so without introducing new issues on Windows.

Reply:
No blocking suggestion. Once you merge it I can create an issue to work on that.